conans/client/graph/graph_binaries.py

```diff
@@ -91,7 +104,11 @@ def _evaluate_node(self, node, build_mode, update, evaluated_references, remote_
         with self._client_cache.package_lock(package_ref):
             if is_dirty(package_folder):
                 output.warn("Package is corrupted, removing folder: %s" % package_folder)
-                rmdir(package_folder)
+                if not node.recipe == RECIPE_EDITABLE:
```

Comment:
I don't understand. Can the execution come here without exiting first at the `if self._client_cache.installed_as_editable(conan_ref):`?

Reply:
Deleted, it was impossible to reach that line. I've put an `assert` just in case, I don't want to remove the user workspace.